examples/gno.land/r/nt/boards2/v1/freeze.gno

```diff
@@ -0,0 +1,76 @@
+package boards2
+
+import "std"
+
+type freezeArgs struct {
+	boardID  BoardID
+	threadID PostID
+	replyID  PostID
+}
+
+func (args freezeArgs) getThread() (*Post, *Board) {
+	board := mustGetBoard(args.boardID)
+	assertBoardIsNotFrozen(board)
+
+	thread := mustGetThread(board, args.threadID)
+	return thread, board
+}
+
+func (args freezeArgs) getReply() (*Post, *Board) {
+	board := mustGetBoard(args.boardID)
+	assertBoardIsNotFrozen(board)
+
+	thread := mustGetThread(board, args.threadID)
+	assertThreadVisible(thread)
+	assertThreadIsNotFrozen(thread)
+
+	reply := mustGetReply(thread, args.replyID)
+	assertReplyVisible(reply)
+	return reply, board
+}
+
+func setReplyReadOnly(args freezeArgs, isReadOnly bool) {
+	reply, board := args.getReply()
+
+	if isReadOnly {
+		// disallow freezing already frozen reply
+		assertReplyIsNotFrozen(reply)
+	}
```

Comment:
This could be moved to `FreezeReply()` as simple assert without condition as it's only needed there

Reply:
The goal of `setReplyReadOnly` to have a logic in one place. Public methods are just helpers as the spec requires to have simple calls.